ggml/src/ggml-cpu/ggml-cpu.c

```diff
@@ -2439,6 +2449,7 @@ static void ggml_init_arm_arch_features(void) {
     uint32_t hwcap2 = getauxval(AT_HWCAP2);
 
     ggml_arm_arch_features.has_neon = !!(hwcap & HWCAP_ASIMD);
+    ggml_arm_arch_features.has_dotprod = !!(hwcap && HWCAP_ASIMDDP);
```

Comment:
why `&&` ... mask need `&` ?

Reply:
This is most definitely a bug, do you want to open a PR to fix it?